### DISCUSSION TITLE
Multi-Document Support for Large Applications

### DISCUSSION THREAD

**Discussion by HaikAsatryan** [scalar/scalar#3496] [Ideas]
Hello Scalar team,

First of all, thank you for building an amazing API documentation tool. However, I am writing to request a key feature that is crucial for large-scale applications: **Multi-Document Support**. Only for that missing feature our team decided not to switch from swagger to scalar in `.net 9`.

In large applications, it's common to manage multiple versions or distinct sets of APIs that cater to different services or environments (e.g., public/private APIs, or multiple versions for different clients). Currently, Scalar requires users to manually change the URL to switch between documents, which is not practical for large-scale projects with multiple API versions.

### Why this feature is essential?
- Scalability: In enterprise-level applications, managing multiple OpenAPI documents is a standard need. Developers should be able to seamlessly switch between versions or different API sets, without manually typing URLs.
- Consistency: Tools like Swagger provide an intuitive UI for multi-document support, allowing easy navigation between versions or services. This feature would bring Scalar up to par with other popular tools.
- User Experience: As applications grow, the ease of navigating between multiple API versions is critical for both internal developers and external users (e.g., third-party integrators). Without this feature, the user experience is hindered, making Scalar less appealing for larger teams and projects.

### Suggested Implementation:
Introduce a dropdown or selector in the UI to allow users to switch between multiple OpenAPI documents.

**Comment by xC0dex** [comment]
Hey @HaikAsatryan, I just spot that this feature request was reported [here](https://github.com/scalar/scalar/issues/1851)!

**Reply by HaikAsatryan** [reply]
Yes, you are right, thank you.
